bftengine/include/bftengine/ReplicaConfig.hpp

```diff
@@ -233,6 +233,10 @@ class ReplicaConfig : public concord::serialize::SerializableFactory<ReplicaConf
                std::chrono::seconds,
                std::chrono::seconds{60},
                "Time interval in seconds to monitor disk usage by db checkpoints");
+  CONFIG_PARAM(dbCheckpointDiskSpaceThreshold,
+               std::string,
+               "0.5",
```

Comment:
This threshold must be less than 1. It should be good if this is kept below 0.8  (which will be a stricter upper bound on the threshold).
Any value greater than this upper bound will be clipped. This can be done one time instead of doing it every time we use this variable. So I prefer that we should take it as a double or float and then add a data validator in the MR. Taking it as a string and converting it to a double everytime can be avoided.

Reply:
these config parameters need to be serializable and we do not have floating point number serialization support in current utility. So keeping it this way for now.